modules/python/math.poly/source/math_poly.i

```diff
@@ -129,6 +147,35 @@ typedef math::linear::Vector<double> VectorDouble;
     {
         return math::poly::TwoD<double>(*$self);
     }
+
+    PyObject* __call__(PyObject* x_input, PyObject* y_input)
+    {
+        if (!PySequence_Check(x_input))
+        {
+            PyErr_SetString(PyExc_TypeError,"Expecting a sequence");
+            return NULL;
+        }
+        if (!PySequence_Check(y_input))
+        {
+            PyErr_SetString(PyExc_TypeError,"Expecting a sequence");
+            return NULL;
+        }
+        int N = PyObject_Length(x_input);
```

Comment:
Changed ‘int’ to ‘Py_ssize_t’ throughout update.

From: Adam Sylvester [mailto:notifications@github.com]
Sent: Tuesday, November 24, 2015 11:36 AM
To: mdaus/coda-oss <coda-oss@noreply.github.com>
Cc: David Starinshak <David.Starinshak@mdaus.com>
Subject: Re: [coda-oss] Extends SWIG bindings for math.poly (#66)


In modules/python/math.poly/source/math_poly.i<http://cp.mcafee.com/d/1jWVIe6xEq6jqb33PbOpEVo76XCSm6hMUQsLLCTNPb3a9EVd79J6XRTD3hPsQsL6QkkkTTCkjhOMrgQP93GjScEk9Wl1iG6JQVsSGxHtendI-yqem7T7-LP1EVoujWZOWqqrf8T7cCzAQsYJteOaaJQn-l3PWApmU6CQjqpK_8KcKnhhshsdTdAVPmEBC4pj9JAenNl3mWpqG6BWQFyPpV-ee0oxaBCWkBm0H5xYs-qem3hOOr8lrxrW01uouohpbT-XNt3jcAeFfoOxgDFkVl3mWsK2G4hfycFASO7bCTPhO-Or1vF6y0QJSGxFuJa6BQQg3gvfU8kd40M0Eq4zh0gB12k2d3h0Xm9EwlSpfBGDBzh0qa0cssd59JZMTWeB>:

> @@ -129,6 +147,35 @@ typedef math::linear::Vector<double> VectorDouble;

>      {

>          return math::poly::TwoD<double>(*$self);

>      }

> +

> +    PyObject* __call__(PyObject* x_input, PyObject* y_input)

> +    {

> +        if (!PySequence_Check(x_input))

> +        {

> +            PyErr_SetString(PyExc_TypeError,"Expecting a sequence");

> +            return NULL;

> +        }

> +        if (!PySequence_Check(y_input))

> +        {

> +            PyErr_SetString(PyExc_TypeError,"Expecting a sequence");

> +            return NULL;

> +        }

> +        int N = PyObject_Length(x_input);

To be pedantic, this should be a Py_ssize_t right? This will make a difference on Windows if you ever had something so long it wouldn't fit in 4 bytes. This should ripple through the for loop on line 170 and other similar spots below.

—
Reply to this email directly or view it on GitHub<http://cp.mcafee.com/d/k-Kr418q6xASyMMYOYCqem1NKVJBxAsed7bXVJYsOMOyqejhOrhKZtVMQsTd7bNJ555dZVB4QsI6QdcOgWAZza52uBgkGxHtendGEqTjBPrfECzBxZN_HYMqem7A-LsKCCCPOdNP9EVd7fbnjIyyHt5_BgY-F6lK1FJcSCrLObzbBQkn4n3tPpesRG9px6kOrp3BYlgRKCmGxFuJaoISuvzzCeOf-w6hov7fCzBwQsICO5mUm-w0nC7C4miZ_KYngQP93GjScEk9WlelgRKDbwGx4jUzapdIxOVJYQsLICMnWhEwdbtGEqnHixFtd40Q7P-253h0c0a6x8Qg49ggB0zgQgeRyq85tCjVqFVoQg6yw3773hirvsdLf8F>.


The information contained in this communication is confidential, is intended only for the use of the recipient(s) named above, and may be legally privileged. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution, or copying of this communication is strictly prohibited. If you have received this communication in error, please re-send this communication to the sender and delete the original message or any copy of it from your computer system.